cert lifecycle fixes (#108)
peppelinux/draft-demarco-acme-openid-federation: draft-demarco-acme-openid-federation.md

# Certificate Lifecycle and Revocation

The identity of the Requestor is verified through proof of possession of a
private key corresponding to a public key attested within a Trust Chain. The

> Trust Chain has an expiration time, and its content MUST not be trusted past the

Reviewer: ```suggestion
Trust Chain has an expiration time, and its content MUST NOT be trusted past the
```

> private key corresponding to a public key attested within a Trust Chain, which
> has an expiration time past which the Trust Chain's content MUST not be trusted
> ({{Section 10.2 of OPENID-FED}}{: relative="#section-10.2"}).

Reviewer 2: rephrasing for clarity:
```suggestion
private key corresponding to a public key attested within a Trust Chain. The Trust Chain has
an expiration time, and its content MUST not be trusted past the expiration time
({{Section 10.2 of OPENID-FED}}{: relative="#section-10.2"}).
```

The `notBefore` and `notAfter` fields of issued certificates MUST represent
dates before the Trust Chain's expiration time. If the Requestor's newOrder
request ({{Section 7.4 of !RFC8555}}) contains `notAfter` or `notBefore` fields

> that make this impossible, the Certificate Issuer MUST return an error.

Reviewer: Seems like we should specify the ACME error here.

Author: I'm inclined to agree, but interestingly ACME itself does not: https://datatracker.ietf.org/doc/html/rfc8555#section-7.4

>    The server MUST return an error if it cannot fulfill the request as
>    specified, and it MUST NOT issue a certificate with contents other
>    than those requested.  If the server requires the request to be
>    modified in a certain way, it should indicate the required changes
>    using an appropriate error type and description.

Reviewer: huh, okaayyyy

Author: ...that said, we could read the second sentence to mean that our draft should provide some guidance here. An error indicating that the X.509 notBefore/notAfter is incompatible with the OIDF exp would do.
